testnet/stacks-node/src/tests/signer/v0.rs

```diff
@@ -10815,3 +10815,448 @@ fn injected_signatures_are_ignored_across_boundaries() {
+    let info_before = get_chain_info(&conf);
+
+    // submit a tx so that the miner will ATTEMPT to mine a stacks block N+3
+    let transfer_tx = make_stacks_transfer(
+        &sender_sk,
+        sender_nonce,
+        send_fee,
+        signer_test.running_nodes.conf.burnchain.chain_id,
+        &recipient,
+        send_amt,
+    );
+    let tx = submit_tx(&http_origin, &transfer_tx);
+    info!("Submitted tx {tx} in attempt to mine block N+2");
+    sender_nonce += 1;
+
+    wait_for(30, || {
+        Ok(signer_test
+            .stacks_client
+            .get_peer_info()
+            .expect("Failed to get peer info")
+            .stacks_tip_height
+            > stacks_height_before
+            && blocks_mined2.load(Ordering::SeqCst) > blocks_processed_before_2
+            && get_chain_info(&conf).stacks_tip_height > info_before.stacks_tip_height)
+    })
+    .expect("Timed out waiting for Miner 2 to Mine Block N+3");
+
+    assert_eq!(get_chain_info(&conf).stacks_tip_height, block_n_height + 3);
+
+    info!("------------------------- Miner 1 Wins the Next Tenure -------------------------");
+
+    let blocks_processed_before_1 = blocks_mined1.load(Ordering::SeqCst);
+    let mined_before = test_observer::get_mined_nakamoto_blocks().len();
+
+    next_block_and(
+        &mut signer_test.running_nodes.btc_regtest_controller,
+        30,
+        || {
+            Ok(
+                blocks_mined1.load(Ordering::SeqCst) > blocks_processed_before_1
+                    && test_observer::get_mined_nakamoto_blocks().len() > mined_before,
+            )
+        },
+    )
+    .expect("Timed out waiting for Miner 1 to Mine Block N+1'");
+
+    let blocks = test_observer::get_mined_nakamoto_blocks();
+    let last_block = blocks.last().expect("No blocks mined");
+    assert_eq!(last_block.stacks_height, block_n_height + 1);
+
+    info!("------------------------- Miner 1 Mines N+2 -------------------------");
+
+    let blocks_processed_before_1 = blocks_mined1.load(Ordering::SeqCst);
+    let mined_before = test_observer::get_mined_nakamoto_blocks().len();
+
+    // submit a tx so that the miner will ATTEMPT to mine a stacks block N+2
+    let transfer_tx = make_stacks_transfer(
+        &sender_sk,
+        sender_nonce,
+        send_fee,
+        signer_test.running_nodes.conf.burnchain.chain_id,
+        &recipient,
+        send_amt,
+    );
+    let tx = submit_tx(&http_origin, &transfer_tx);
+    info!("Submitted tx {tx} in attempt to mine block N+2");
+
+    wait_for(30, || {
+        Ok(
+            blocks_mined1.load(Ordering::SeqCst) > blocks_processed_before_1
+                && test_observer::get_mined_nakamoto_blocks().len() > mined_before,
+        )
+    })
+    .expect("Timed out waiting for Miner 1 to Mine Block N+2");
+
+    let blocks = test_observer::get_mined_nakamoto_blocks();
+    let last_block = blocks.last().expect("No blocks mined");
+    assert_eq!(last_block.stacks_height, block_n_height + 2);
```

Comment:
Because miner 2 produced block N+2 and N+3, would that mean that miner 2's chain tip is inconsistent with miner 1's chain tip?

Reply:
Yes, but I think with the way things are now, if miner 1 wins the next tenure, it would mine off of miner 2's tip, since it is higher. I will verify that with some additions to this test. I think ultimately, signers should not allow a reorg if there is more than 1 block in the tenure that is being reorged, then we avoid this scenario. I will open an issue to discuss that further.

Reply:
I forgot, I already opened that issue - https://github.com/stacks-network/stacks-core/issues/5684.

Reply:
Until #5684 gets addressed, we're effectively allowing Miner 1 to completely orphan Miner 2's blocks during its tenure?  This should be documented in the changelog.

Reply:
Sure, I can add something, but that is not a new policy. It was always the case that the signers would allow this reorg. The change here is that the miner will actually build multiple blocks in the reorging tenure, instead of just the one.

Reply:
I'll wait for #5724 to merge before editing the changelog since it will just cause a conflict.

Reply:
Added to changelog in 13aaa3f68